### PR TITLE
Port to SDL2

### PR DESCRIPTION
**Pull request by sharkwouter**
The SDL2 port was done by EXL. I fixed a bug and made building on Windows work with CMake. I also updated the README to describe how to develop on Windows and removed all unused files. Please do a quick test of this when reviewing this.

### CODE REVIEW

**Comment by sharkwouter** [comment]
This is now ready to be reviewed. I managed to figure out the problems I was having on Windows and I fixed the crash I was running into. If this is not seen as spam, could a label with the name "hacktoberfest-accepted" be added to this PR? I am trying to earn a t-shirt :)

**Comment by sharkwouter** [comment]
@Asmageddon @bobvanderlinden

File: src/game/options.c [hunk 1]
```diff
@@ -781,11 +782,13 @@ void videooptionsmenu(void)
       SDL_GL_SetAttribute(SDL_GL_BLUE_SIZE,8);
       SDL_GL_SetAttribute(SDL_GL_ALPHA_SIZE,8);
       }
-  
-    if (windowinfo.fullscreen)
-      SDL_SetVideoMode(windowinfo.resolutionx,windowinfo.resolutiony,windowinfo.bitsperpixel,SDL_OPENGL|SDL_FULLSCREEN);
-    else
-      SDL_SetVideoMode(windowinfo.resolutionx,windowinfo.resolutiony,windowinfo.bitsperpixel,SDL_OPENGL);
+
+    printf( "options.c Opening screen %dx%dx%d\n", windowinfo.resolutionx, windowinfo.resolutiony, windowinfo.bitsperpixel );
```
Comment (bobvanderlinden) [inline src/game/options.c:786]: Debug log may be removed.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -123,24 +123,42 @@ set(GISH_H
...
 endif(WIN32)
 
 if(APPLE)
     list(APPEND GISH_SRCS sdl/SDLMain.m)
     list(APPEND GISH_H sdl/SDLMain.h)
 endif(APPLE)
 
-add_executable(freegish ${GISH_SRCS} ${GISH_H})
+if(WIN32)
+    add_executable(freegish WIN32 ${GISH_SRCS} ${GISH_H})
```
Comment (bobvanderlinden) [inline src/CMakeLists.txt:135]: WIN32 is not needed. It'll build a Windows executable without this as well.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -123,24 +123,42 @@ set(GISH_H
...
+        ${OPENGL_LIBRARIES}
+        SDL2::SDL2
+        SDL2::SDL2main
+        Ogg::ogg
+        Vorbis::vorbis
+        Vorbis::vorbisenc
+        Vorbis::vorbisfile
+        png
+        OpenAL::OpenAL
+    )
```
Comment (bobvanderlinden) [inline src/CMakeLists.txt:151]: The method used for Linux is also possible to be used for Windows if `find_package` without `CONFIG` is used.

File: CMakeLists.txt [hunk 4]
```diff
@@ -16,15 +16,30 @@ endif(NOT ${INSTALL_FHS})
...
+    find_package(Vorbis CONFIG REQUIRED)
+    find_package(libpng CONFIG REQUIRED)
+    find_package(SDL2 CONFIG REQUIRED)
+    find_package(OpenAL CONFIG REQUIRED)
+    find_package(OpenGL REQUIRED)
+
+else()
+    # Find SDL2 with pkg-config
+    include(FindPkgConfig)
+    pkg_search_module(SDL2 REQUIRED sdl2)
```
Comment (bobvanderlinden) [inline CMakeLists.txt:30]: Using `find_package(SDL2 REQUIRED)` also seems to work on my Windows/Linux machines.

File: CMakeLists.txt [hunk 5]
```diff
@@ -16,15 +16,30 @@ endif(NOT ${INSTALL_FHS})
...
-find_package(Vorbis REQUIRED)
-find_package(PNG REQUIRED)
-find_library(M_LIBRARIES m)
+if(MSVC)
+    find_package(Ogg CONFIG REQUIRED)
+    find_package(Vorbis CONFIG REQUIRED)
+    find_package(libpng CONFIG REQUIRED)
+    find_package(SDL2 CONFIG REQUIRED)
+    find_package(OpenAL CONFIG REQUIRED)
+    find_package(OpenGL REQUIRED)
```
Comment (bobvanderlinden) [inline CMakeLists.txt:25]: Using `find_package` without `CONFIG` is also possible on both Windows and Linux, so the `if(MSVC)` can be removed.

**Review by bobvanderlinden — COMMENTED** [review]
Nice! Looks good. I was able to build this on Windows. On Linux I needed to do some changes to workaround some warnings, like using legacy OpenGL instead of modern.

While working on this I found that most of the cmake configuration can be aligned between Windows and Linux without the if conditions. It isn't really finished, but you can see the results here:

https://github.com/bobvanderlinden/freegish/commit/cc61d827db50fd4aaf157cec18f0f031093d61c1

File: src/CMakeLists.txt (hunk 2 above)
Comment (sharkwouter) [reply src/CMakeLists.txt:135]: Only if you build with Visuall Studio by hand. If you run `cmake --build .` to build WIN32 is needed here.

File: src/CMakeLists.txt (hunk 3 above)
Comment (sharkwouter) [reply src/CMakeLists.txt:151]: I did a test on Ubuntu 22.04 with the packages from the repository. They do not provide the Vorbis, OpenAL and Ogg link targets. find_package doesn't work with Ogg at all. Not sure if linking to ogg is actually required in Windows, I'll look into that, the README could have been wrong.

File: CMakeLists.txt (hunk 4 above)
Comment (sharkwouter) [reply CMakeLists.txt:30]: I tested this on Ubuntu 22.04 and this seems to work. I'll change it.

**Comment by sharkwouter** [comment]
I did some minor improvements now. This is not ready for another review yet, though. I need to test some things on Windows first.

File: src/CMakeLists.txt (hunk 2 above)
Comment (bobvanderlinden) [reply src/CMakeLists.txt:135]: Ah I wasn't aware of this method :+1:

File: src/CMakeLists.txt (hunk 3 above)
Comment (bobvanderlinden) [reply src/CMakeLists.txt:151]: I tried both Linux (NixOS) and Windows (using vcpkg), both work fine without Ogg. Music and sound effects are still working :+1:

**Comment by bobvanderlinden** [comment]
I've also played around some more trying to simplify the cmake files. I ended up with this patch:

https://github.com/bobvanderlinden/freegish/commit/62f10d6f5e8cdd9536153aaa578ec45fa5fd45fb

**Comment by sharkwouter** [comment]
@bobvanderlinden I'd say merge it like this and build on top of it then. I have thought about it for a while and I think that will probably work out better.

**Review by bobvanderlinden — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**